### PR TITLE
Animal create/update features

### PR DESCRIPTION
**Pull request by saulyz**
WIP. Before the mutations

### CODE REVIEW

**Comment by saulyz** [comment]
seems like a dirty source branch... Some irrelevant commits are pulled from the history

File: src/components/animal/create/AnimalForm.tsx [hunk 1]
```diff
@@ -1,35 +1,70 @@
...
+        return <p>Error!</p>;
+    }
+
+    const onSubmit = (formData: AnimalFormData) => {
+        // filter/cleanup RichTextEditorField
+        formData.comments = filterRichTextEditorField(formData.comments);
+
+        // convert birthDate string to TS
+        if (formData.details && formData.details.birthDate) {
+            formData.details.birthDate = filterBirthDateField(formData.details?.birthDate);
```
Comment (rikius) [inline src/components/animal/create/AnimalForm.tsx:58]: not seeing why we need to change birthDate here.

File: src/components/animal/create/AnimalForm.tsx [hunk 2]
```diff
@@ -1,35 +1,70 @@
...
+    });
+    const methods = useForm({ defaultValues: Number(id) ? data?.animal : DEFAULT_VALUES });
+    const { handleSubmit, reset } = methods;
     const formRef = useRef<HTMLFormElement>(null);
 
-    const onSubmit = (data: AnimalFormData) => {
+    useEffect(() => {
+        if (data?.animal) {
+            // convert birthDate TS to string
+            const animalCopy = Object.assign({}, data.animal, {
```
Comment (rikius) [inline src/components/animal/create/AnimalForm.tsx:35]: not seeing why we need to convert data here.

File: src/components/animal/create/DetailsStep.tsx [hunk 3]
```diff
@@ -81,30 +102,17 @@ function DetailsStep() {
                 </Grid>
                 <Grid item xs={12} sm={6}>
                     <TextInput
-                        type="date"
                         name="details.birthDate"
```
Comment (rikius) [inline src/components/animal/create/DetailsStep.tsx:105]: we can convert data to correct view format here.

**Review by rikius — CHANGES_REQUESTED** [review]

**Comment by saulyz** [comment]
waiting for the Backend team changes regarding type structures

File: src/components/animal/create-update/AnimalForm.tsx [hunk 4]
```diff
@@ -1,78 +1,39 @@
...
 
     useEffect(() => {
-        if (data?.animal) {
-            // convert birthDate TS to string
-            const animalCopy = Object.assign({}, data.animal, {
-                details: { ...data.animal.details, birthDate: getYMDDateFromTS(data.animal.details?.birthDate) },
-            });
-            reset(animalCopy);
+        if (animal) {
+            reset(getDefaultFormValues(animal));
```
Comment (rikius) [inline src/components/animal/create-update/AnimalForm.tsx:23]: remember why we need to have this reset?

**Review by rikius — APPROVED** [review]

**Review by rikius — APPROVED** [review]

**Pull request merged**